src/DynamoCore/Core/NotificationObject.cs

```diff
@@ -20,6 +22,11 @@ public abstract class NotificationObject : INotifyPropertyChanged
         /// <param name="propertyName">The property that has a new value.</param>
         protected virtual void RaisePropertyChanged(string propertyName)
         {
+            //TODO profile this?
```

Comment:
I see O(n) worst-case complexity as it would need to search for the property in `PropertyNames`.